processor/metricstransformprocessor/metrics_transform_processor.go

```diff
@@ -148,6 +152,34 @@ func (mtp *metricsTransformProcessor) ProcessMetrics(_ context.Context, md pdata
 		for _, transform := range mtp.transforms {
 			matchedMetrics := transform.MetricIncludeFilter.getMatches(nameToMetricMapping)
 
+			if transform.Action == Group && len(matchedMetrics) > 0 {
+				// create new ResouceMetrics bucket
```

Comment:
Please move the Group functionality into its own method. All other operations transform data in called methods instead of inside this method. It needs to be consistent.

Reply:
Thank you! Will push this logic into its own method.

Reply:
Ok pushed the new changes. Kindly let me know.